### DISCUSSION TITLE
Checking communication loss

### DISCUSSION THREAD

**Discussion by LaptosPirose** [ottowayi/pycomm3#290] [Q&A]
I am using pycomm3 to communicate with a Logix PLC and everything is working weel, but how can i do to check communication all time and don't crash the programm?
When I test te continuous communication and take off the Ethernet cable, the programm stop. How can I check this e try to reestart the communication? The simple part of programm is below and my attempt to check datas withi nosuccess.

from pycomm3 import LogixDriver
class Connect():
    def __init__(self, plc_ip : str):
        self.__plc_ip = plc_ip
        self.__plc = self.__plc_connect()
    
    @property
    def get_plc_ip(self):
        return self.__plc_ip
    @property
    def get_plc_connected(self):
        if self.__plc != 'failed':
            __plc_connected = LogixDriver(self.__plc)
            return __plc_connected
        else:
            return 'failed'
    def __plc_connect(self):
        plc = LogixDriver(self.__plc_ip)
        try:
            __plcmap = plc.discover()
            __plc_encontrado = False
            for __i in __plcmap:
                if (self.__plc_ip == __i['ip_address']):
                    print(f"PLC de IP {__i['ip_address']} existe!" )
                    __plc_encontrado = True
                    __ip_encontrado = __i['ip_address']
            if not __plc_encontrado:
                    print(f'PLC de IP {self.__plc_ip} não encontrado')
                    return 'failed'
            return __ip_encontrado
        except ConnectionError:
            print('Tentativa de conexão falhou!')

def __main__():
    from rockwell_modules.class_connect import Connect as Comm

    acople = Comm('192.168.0.96')
    acople = acople.get_plc_connected
    
    if acople != 'failed':
        acople.open()
        print(acople)
        for i in range(1,1000):
            try:
                print(acople.read('DKZ1_Numero_Falha_ETB').value)
            except ConnectionError:
                acople.close()
if __name__ == '__main__':
    __main__()

**Comment by ottowayi** [comment]
I think if you broaden your `except` block to `except Exception` it should work better, `ConnectionError` is not one that I think is raised by pycomm3.  When you catch the error, you could try to reconnect or choose to exit.
